bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -219,16 +220,20 @@ public Void run() {
         @Override
         public void thingUpdated(final Thing thing) {
             thingUpdatedLock.add(thing.getUID());
-            AccessController.doPrivileged(new PrivilegedAction<Void>() {
+            Thing ret = AccessController.doPrivileged(new PrivilegedAction<Thing>() {
 
                 @Override
-                public Void run() {
-                    managedThingProvider.update(thing);
-                    return null;
+                public Thing run() {
+                    return managedThingProvider.update(thing);
                 }
 
             });
             thingUpdatedLock.remove(thing.getUID());
+            if (ret == null) {
+                throw new IllegalStateException(
+                        MessageFormat.format("Could not update thing {}. Most likely because it is read-only.",
```

Comment:
Seems that this needs to be "Could not update thing **{0}**..."
Else a `java.lang.IllegalArgumentException: can't parse argument number: ` is thrown.